engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/registrar/EndNodeInterpretationFunction.scala

```diff
@@ -0,0 +1,28 @@
+package pl.touk.nussknacker.engine.process.registrar
+
+import org.apache.flink.api.common.functions.{RichMapFunction, RuntimeContext}
+import org.apache.flink.configuration.Configuration
+import pl.touk.nussknacker.engine.api.{Context, MetaData, ProcessListener}
+
+private[registrar] class EndNodeInterpretationFunction(prepareListeners: RuntimeContext => Seq[ProcessListener],
```

Comment:
I think it suffices to use ProcessPartFunction? It opens only needed services. Or at least let's make sth similar to ExceptionHandlerFunction?